pkg/tools/gen/genkcl.go

```diff
@@ -0,0 +1,169 @@
+package gen
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"kusionstack.io/kclvm-go/pkg/logger"
+)
+
+// TODO: is need to parse from Comment. Now get kcl schema from Tag
+type GenKclOptions struct {
+	KclPackage       string
+	ParseFromComment bool
+	ParseFromTag     bool
+}
+
+type kclGenerator struct {
+	opts *GenKclOptions
+}
+
+// GenKcl translate go struct to kcl schema code.
+func GenKcl(w io.Writer, filename string, src interface{}, opts *GenKclOptions) error {
+	return newKclGenerator(opts).GenSchema(w, filename, src)
+}
+
+func newKclGenerator(opts *GenKclOptions) *kclGenerator {
+	if opts == nil {
+		opts = new(GenKclOptions)
+	}
+	return &kclGenerator{
+		opts: opts,
+	}
+}
+
+func (k *kclGenerator) GenSchema(w io.Writer, filename string, src interface{}) error {
+	fmt.Fprintln(w)
+	goStructs, err := ParseGoSourceCode(filename, src)
+	if err != nil {
+		return err
+	}
+	for _, goStruct := range goStructs {
+		fmt.Fprintf(w, "schema %s:\n", goStruct.Name)
+		if goStruct.StructComment != "" {
+			fmt.Fprintf(w, "\"\"\"%s\"\"\"\n", goStruct.StructComment)
+		}
+		for _, field := range goStruct.Fields {
+			kclFieldName, kclFieldType, err := k.parserGoStructFieldTag(field.FieldTag)
+			if err != nil {
+				logger.GetLogger().Warningf("get struct tag key kcl info err: %s, will jump the struct field, field info %#v", err.Error(), field)
```

Comment:
add the log warning handle the err, it will jump if a struct field don't have kcl tag.

Reply:
![image](https://user-images.githubusercontent.com/42363586/230539485-9d186c26-30fb-4cf4-b16b-b368f26984d2.png)